src/Compilers/CSharp/Portable/Symbols/TypeUnification.cs

```diff
@@ -287,9 +287,10 @@ private static bool Contains(TypeSymbol type, TypeParameterSymbol typeParam)
                         NamedTypeSymbol namedType = (NamedTypeSymbol)type;
                         while ((object)namedType != null)
                         {
-                            foreach (TypeSymbol typeArg in namedType.TypeArgumentsNoUseSiteDiagnostics)
+                            ImmutableArray<TypeSymbol> typeParts = namedType.IsTupleType ? namedType.TupleElementTypes : namedType.TypeArgumentsNoUseSiteDiagnostics;
+                            foreach (TypeSymbol typePart in typeParts)
                             {
-                                if (Contains(typeArg, typeParam))
+                                if (Contains(typePart, typeParam))
```

Comment:
They are extremely similar indeed.
The one difference that I could spot is that `TypeSymbolExtensions.ContainsTypeParamater` relies on `VisitType` which doesn't dig into error types, whereas `TypeUnification.Contains` does.

Given that, it's tempting to re-use `TypeSymbolExtensions.ContainsTypeParameter`. What do you think?